### PR TITLE
Remove deprecated annotations from the io.quarkus.arc.config package

### LINKED ISSUE #22101: Remove the deprecated io.quarkus.arc.config.ConfigProperties
### Description

The annotations from the `io.quarkus.arc.config` package have been deprecated for a long time. Users are encouraged to use the `io.smallrye.config.ConfigMapping` instead. I think it's time to remove them completely!

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by mkouba**
- removed ConfigIgnore, ConfigPrefix and ConfigProperties
- removed the ArcConfig#configPropertiesDefaultNamingStrategy
- removed the relevant section in the docs
- the necessary logic from the io.quarkus.arc.deployment.configproperties package was move into the io.quarkus.arc.deployment.configproperties module
- resolves #22101

### CODE REVIEW

**Comment by mkouba** [comment]
> the necessary logic from the io.quarkus.arc.deployment.configproperties package was move into the io.quarkus.arc.deployment.configproperties module

Note that I only did minimal changes to the logic so it might include a lot of useless stuff. However, it's fairly complex and I didn't want to spend a lot of time with refactoring. We may improve this part later in a separate pull request.

File: extensions/config-yaml/deployment/src/test/resources/configprops.yaml
Comment (mkouba) [inline extensions/config-yaml/deployment/src/test/resources/configprops.yaml:8]: Support for list of objects in YAML was removed in https://github.com/quarkusio/quarkus/commit/ab40883447ea89ae4334c371c315b954ba5f7b5a.

**Comment by mkouba** [comment]
Native Tests - Main fails with:
```
ERROR: Failed to start application (with profile prod)
java.lang.IllegalArgumentException: SRCFG00013: No Converter registered for class java.time.Instant
	at io.smallrye.config.SmallRyeConfig.requireConverter(SmallRyeConfig.java:484)
	at io.smallrye.config.ConfigMappingContext.getConverter(ConfigMappingContext.java:113)
	at io.smallrye.config.ConfigMappingContext.lambda$getValueConverter$4(ConfigMappingContext.java:92)
	at java.util.HashMap.computeIfAbsent(HashMap.java:1220)
	at io.smallrye.config.ConfigMappingContext.getValueConverter(ConfigMappingContext.java:89)
	at io.quarkus.it.config.ConfigPropertiesResource$GreetingConfigurationI721028930Impl.<init>(Unknown Source)
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at io.smallrye.config.ConfigMappingLoader.configMappingObject(ConfigMappingLoader.java:65)
	at io.smallrye.config.ConfigMappingContext.constructGroup(ConfigMappingContext.java:80)
	at io.smallrye.config.ConfigMappingContext.constructRoot(ConfigMappingContext.java:76)
	at io.smallrye.config.ConfigMappingProvider.mapConfiguration(ConfigMappingProvider.java:940)
	at io.smallrye.config.ConfigMappingProvider.lambda$mapConfiguration$3(ConfigMappingProvider.java:923)
	at io.smallrye.config.SecretKeys.lambda$doUnlocked$0(SecretKeys.java:20)
	at io.smallrye.config.SecretKeys.doUnlocked(SecretKeys.java:29)
	at io.smallrye.config.SecretKeys.doUnlocked(SecretKeys.java:19)
	at io.smallrye.config.ConfigMappingProvider.mapConfiguration(ConfigMappingProvider.java:923)
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:450)
	at io.quarkus.runtime.generated.Config.readConfig(Unknown Source)
	at io.quarkus.deployment.steps.RuntimeConfigSetup.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:110)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:69)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
```

@radcortez does it ring a bell?

**Comment by mkouba** [comment]
It seems that there is no built-in converter for `LocalDate` but the test works in jvm mode...

**Comment by radcortez** [comment]
Let me know if you want me to look into it.

**Comment by radcortez** [comment]
Its awesome to see so much code removed :)

File: core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigMappingUtils.java
Comment (mkouba) [inline core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigMappingUtils.java:116]: @radcortez this should fix the problem with `Optional`. It's not a robust solution but I have no idea whether it's needed for other containers (such as `List`).

Comment (radcortez) [reply core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigMappingUtils.java:116]: Hum, yes, we probably need it for List as well.

Comment (mkouba) [reply core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigMappingUtils.java:116]: How about `Set` and others? Should I file a separate issue?

Comment (mkouba) [reply core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigMappingUtils.java:116]: https://github.com/quarkusio/quarkus/issues/28065

**Comment by mkouba** [comment]
The JVM Tests - JDK 18 failure is ` Failed to execute goal io.fabric8:docker-maven-plugin:0.40.2:start (docker-start) on project quarkus-integration-test-reactive-oracle-client: I/O Error: [docker.io/gvenzl/oracle-xe:21-slim-faststart]: Timeout after 60164 ms `.

The Quickstarts Compilation failure looks related to the Jandex 3.0 upgrade? ` Build step io.quarkus.deployment.steps.ApplicationIndexBuildStep#build threw an exception: java.lang.NoSuchMethodError: 'void org.jboss.jandex.Indexer.index(java.io.InputStream)'` CC @Ladicek

**Comment by mkouba** [comment]
I did rebase the branch for the last time just to be sure ;-)

**Comment by Ladicek** [comment]
I would have hoped that the Quickstarts Compilation problem is fixed by the exclusion added here: https://github.com/quarkusio/quarkus-quickstarts/pull/1180 :thinking:

**Comment by mkouba** [comment]
I don't think the CI failures are related. @radcortez I'm waiting for your approval (or rejection ;-)

**Review by radcortez — APPROVED** [review]

**Comment by radcortez** [comment]
> I don't think the CI failures are related. @radcortez I'm waiting for your approval (or rejection ;-)

Approve of course :)

**Comment by mkouba** [comment]
I've added a brief [note to the migration guide](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.14#configproperties-was-removed).

**Pull request merged**
